test/unit/attributes.js

```diff
@@ -1107,6 +1107,48 @@ QUnit.test( "val(select) after form.reset() (Bug #2551)", function( assert ) {
 	jQuery( "#kk" ).remove();
 } );
 
+QUnit.test( "select.val(space characters) (gh-2978)", function( assert ) {
+	assert.expect( 15 );
+
+	var $select = jQuery( "<select/>" ).appendTo( "#qunit-fixture" ),
+		spaces = {
+			"\\t": {
+				html: "&#09;",
+				val: "\t"
+			},
+			"\\n": {
+				html: "&#10;",
+				val: "\n"
+			},
+			"\\r": {
+				html: "&#13;",
+				val: "\r"
+			},
+			"\\f": "\f",
+			"space": " "
+		},
+		html = "";
+	jQuery.each( spaces, function( key, obj ) {
+		var value = obj.html || obj;
+		html += "<option value='bar" + value + "'>bar" + value + "</option>";
+		html += "<option value='te" + value + "st'>te" + value + "st</option>";
+		html += "<option value='" + value + "baz'>" + value + "baz</option>";
+	} );
+	$select.html( html );
+
+	jQuery.each( spaces, function( key, obj ) {
+		var val = obj.val || obj;
+		$select.val( "bar" + val );
+		assert.equal( $select.val(), "bar", "Value ending with space character (" + key + ") selected" );
```

Comment:
Correct. https://html.spec.whatwg.org/multipage/forms.html#concept-option-value specifies the `value` IDL attribute as the `value` content attribute, falling back on the [`text` IDL attribute](https://html.spec.whatwg.org/multipage/forms.html#dom-option-text), which is the element's content, concatenated, with whitespace stripped and collapsed.